java/src/jmri/jmrix/AbstractMRTrafficController.java

```diff
@@ -1094,6 +1098,38 @@ public void run() {
     }  // end XmtNotifier
 
     /**
+     * Terminate the receive and transmit threads.
+     *<p>
+     * This is intended to be used only by testing subclasses.
+     */
+    public void terminateThreads() {
+        threadStopRequest = true;
+        if (xmtThread != null) {
+            xmtThread.stop();
```

Comment:
True, but the flag does prevent the run method from looping if there is some other error that should terminate the thread, like the input stream no longer being available, which is the issue that lead to #3548.